src/control/controlobject.cpp

```diff
@@ -122,7 +122,15 @@ void ControlObject::setParameterFrom(double v, QObject* pSender) {
 void ControlObject::set(const ConfigKey& key, const double& value) {
     QSharedPointer<ControlDoublePrivate> pCop = ControlDoublePrivate::getControl(key);
     if (pCop) {
-        pCop->set(value, NULL);
+        pCop->set(value, nullptr);
+    }
+}
+
+void ControlObject::toggle(const ConfigKey& key) {
+    QSharedPointer<ControlDoublePrivate> pCop = ControlDoublePrivate::getControl(key);
+    if (pCop) {
+        pCop->set(1, nullptr);
```

Comment:
I believe the intended use is pressing a button which involves pressing in the button (set to 1) and then releasing it (set to 0).
But what if it was already set to 1? Set to 1 and then 0 does not complete a button press then.

Reply:
Indeed a valid concern in theory, but what is the case where you hold a button and try to press it? I'd rather focus on the pragmatic side here.